server/api/post/post.controller.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var _ = require('lodash');
+var Post = require('./post.model');
+var User = require('../user/user.model');
+var Project = require('../project/project.model');
+
+// Get the posts corresponding to a project
+exports.showByProject = function(req, res) {
+  Post.find({ "projectId": req.params.projectId })
+  .populate('author')
+  .exec(function (err, posts) {
+    if(err) { return handleError(res, err); }
+    return res.json(200, posts);
+  });
+};
+
+// Get list of posts
+exports.index = function(req, res) {
+  Post.find({ project: req.params.project }, function (err, posts) {
+    if(err) { return handleError(res, err); }
+    return res.json(200, posts);
+  });
+};
+
+// Get a single post
+exports.show = function(req, res) {
+
+  Post.findById(req.params.id)
+  .populate('author')
+  .exec(function (err, post) {
+    if(err) { return handleError(res, err); }
+    if(!post) { return res.send(404); }
+    return res.json(post);
+  });
+};
+
+// Creates a new post in the DB.
+exports.create = function(req, res) {
+
+  req.body.author = req.user._id;
+  if (!req.body.projectId){ return res.status(400).send("Project not set"); }
+  Project.findOne({'_id': req.body.projectId }, function (err, project) {
```

Comment:
This works but for future reference findById is built into Mongoose and does the same query with a little less code and arguably more clarity. http://mongoosejs.com/docs/api.html#model_Model.findById